source/server/server.cc

```diff
@@ -524,7 +524,10 @@ void InstanceImpl::run() {
   // startup (see RunHelperTest in server_test.cc).
   const auto run_helper =
       RunHelper(*this, options_, *dispatcher_, clusterManager(), access_log_manager_, init_manager_,
-                overloadManager(), [this] { startWorkers(); });
+                overloadManager(), [this] {
+                  dispatcher_->post([this] { notifyCallbacksForStage(Stage::PostInit); });
```

Comment:
Do you need to do a dispatcher post here? I think the enclosing lambda runs on the main dispatcher thread already in which case you can call notifyCallbacksForStage directly.

Reply:
Yep, good point. Updated @eziskind